lib/main.dart

```diff
@@ -115,12 +119,75 @@ Future<void> main() async {
   );
 }
 
-class App extends ConsumerWidget {
+class App extends ConsumerStatefulWidget {
   const App({super.key});
 
+  @override
+  ConsumerState<App> createState() => AppState();
+}
+
+class AppState extends ConsumerState<App> with WidgetsBindingObserver {
+  static final GlobalKey<NavigatorState> rootNavigatorKey =
+      GlobalKey<NavigatorState>();
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    didChangeAppLifecycleStateAsync(state);
+  }
+
+  void didChangeAppLifecycleStateAsync(AppLifecycleState state) async {
```

Comment:
Because it needs to be aware of the app state from whatever screen and not only the splash screen. I did some tests and I noticed a bug that once the Splash screen disappears, the previous methods don't work anymore because the splash screen is not in the navigation stack anymore.